brutils/pis.py

```diff
@@ -105,7 +78,12 @@ def is_valid(pis: str) -> bool:
     True
 
     """
-    return isinstance(pis, str) and validate(pis)
+
+    if isinstance(pis, str) and len(pis) == 11 and pis.isdigit():
+        return pis[-1] == str(_checksum(pis[:-1]))
+
+    else:
+        return False
```

Comment:
Eu trocaria a ordem do código para garantir uma melhor leitura. Primeiro a gente trata a lógica mais genérica. Porque senão for string, por exemplo, nem precisa continuar a validar mais nada. Pode sair. 

Ai depois já sabemos que temos os pré-requisitos básicos garantidos e podemos verificar a lógica "principal", a lógica mais específica.

Outra mudança que sugeri é tirar o else. O else não precisa ser colocado já que tem um return antes no fluxo do if. Assim, quando entrar no if, nunca vai prosseguir para frente, no return já vai sair direto do método. Nunca vai passar pelo else. 

```suggestion
    if isinstance(pis, str) and len(pis) == 11 and pis.isdigit():
        return False

    return  pis[-1] == str(_checksum(pis[:-1]))
```